internal/lang/funcs/filesystem.go

```diff
@@ -69,7 +68,7 @@ func MakeFileFunc(baseDir string, encBase64 bool) function.Function {
 // As a special exception, a referenced template file may not recursively call
 // the templatefile function, since that would risk the same file being
 // included into itself indefinitely.
-func MakeTemplateFileFunc(baseDir string, funcsCb func() map[string]function.Function) function.Function {
+func MakeTemplateFunc(baseDir string, funcsCb func() map[string]function.Function, fileInput bool) function.Function {
```

Comment:
I was wondering why we chose to do it this way?

I think overall it may be best in it's own file, even if there is some duplicate logic. This could help us with the maintainability in the long run.

Right now, as an end user who would look for the templatestring function, I would be confused to find it in filesystem.go, and to have file logic in it.

I propose we move it out to it's own function and put this back as it was.

Reply:
For this I was thinking of two options either creating new file named templatestring.go containing all the templatestring function's logic in it or string.go file is already in opentofu so i can add templatestring function in that file as templatefile function is present in filesystem coz it is related to files so similarly templatestring can be in string  file as it is related to string. which one do you suggest?

Reply:
My personal preference would be to place this alongside other string based methods inside `string.go`.

Reply:
commit #[4fb1950](https://github.com/opentofu/opentofu/pull/1223/commits/4fb1950bb377f66c61af3af40afb85f8d11d343a) resolves this